validator_test.go

```diff
@@ -266,6 +273,31 @@ func (ms *MySuite) TestBadParameter(c *C) {
 	c.Assert(errs["C"], HasError, validator.ErrBadParameter)
 }
 
+func (ms *MySuite) TestValidSliceOfStructs(c *C) {
+	//this should fail because TestStruct is not valid
+	s := TestStructSlice{E: 1, Required: []TestStruct{TestStruct{}}}
+	err := validator.Validate(s)
+	c.Assert(err, NotNil)
+	errs, ok := err.(validator.ErrorMap)
+	c.Assert(ok, Equals, true)
+
+	c.Assert(errs["Required"], HasLen, 1)
+	tsErrs, ok := errs["Required"][0].(validator.ErrorMap)
+	c.Assert(ok, Equals, true)
+	c.Assert(tsErrs["A"], HasError, validator.ErrZeroValue)
+
+	s = TestStructSlice{E: 2, Required: []TestStruct{TestStruct{}, TestStruct{}}, Optional: []TestStruct{TestStruct{}}}
+	err = validator.Validate(s)
+	c.Assert(err, NotNil)
+	errs, ok = err.(validator.ErrorMap)
+	c.Assert(ok, Equals, true)
+	c.Assert(errs["Required"], HasLen, 2)
```

Comment:
Can you add a test here to make sure that `TestStructSlice{E: 3, Required: []TestStruct{}}` fails because the slice contains 0 elements? Also can you add a test `TestStructSlice{E: 3, Required: nil}` to make sure that errors too?